privacyIDEA Authenticator/Endpoint.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  Endpoint.swift
+//  privacyIDEA Authenticator
+//
+//  Created by Nils Behlen on 29.04.19.
+//  Copyright © 2019 Nils Behlen. All rights reserved.
+//
+
+import Foundation
+
+class Endpoint: NSObject, URLSessionDelegate {
+    private var url: String
+    private var callback: EndpointCallback
+    private var data: [String : String]
+    private var sslVerify: Bool
+    private var token: Token
+    
+    init(url: String, data: [String : String], sslVerify: Bool, token: Token, callback: EndpointCallback) {
+        self.url = url
+        self.data = data
+        self.sslVerify = sslVerify
+        self.callback = callback
+        self.token = token
+    }
+    
+    func connect() {
+        U.log("Connecting to \(url)")
+        DispatchQueue.global(qos: .background).async {
+            
+            let url = URL(string: self.url)!
+            var session: URLSession
+            
+            if !self.sslVerify {
+                session = URLSession(configuration: .default, delegate: self, delegateQueue: nil) }
+            else {
+                session = URLSession(configuration: .default, delegate: nil, delegateQueue: nil) }
+            
+            var request = URLRequest(url: url)
+            // timeout is 60s by default
+            request.httpMethod = "POST"
+            
+            do {
+                request.httpBody = try JSONSerialization.data(withJSONObject: self.data, options: .prettyPrinted)
+            } catch let error {
+                U.log("parameters could not be serialized: \(error.localizedDescription)")
+            }
+            request.addValue("application/json", forHTTPHeaderField: "Content-Type")
+            request.addValue("application/json", forHTTPHeaderField: "Accept")
+            
+            // create dataTask using the session object to send data to the server
+            let task = session.dataTask(with: request as URLRequest, completionHandler: { data, response, error in
+                
+                guard error == nil else {
+                    U.log("error while sending:")
+                    self.token.setLastestError(error)
+                    self.callback.errorOccured(self.token)
+                    return
+                }
+                
+                guard let data = data else {
+                    U.log("data error")
```

Comment:
I do not understand this. If data can not be assigned to data? then an error is logged and we return?
What does ``data = data`` imply?

Reply:
It is shadowing + optional unwrapping. The shadowing is unneccessary, i will rename it.